pkg/api/2018-09-30-preview/api/types_test.go

```diff
@@ -177,14 +173,21 @@ func TestUnmarshal(t *testing.T) {
 func TestStructTypes(t *testing.T) {
 	// AgentPoolProfile and MasterPoolProfile types should be identical bar
 	// `Role AgentPoolProfileRole` in the former
+	// MasterPoolProfile has removed OsType, Name
 	app := reflect.TypeOf(AgentPoolProfile{})
 	mpp := reflect.TypeOf(MasterPoolProfile{})
-	if app.NumField() != mpp.NumField()+1 {
+	// Add 3 for Role,OsType,Name which are missing from MasterPoolProfile
+	if app.NumField() != mpp.NumField()+3 {
 		t.Fatalf("mismatch in number of fields: %d vs %d", mpp.NumField(), app.NumField())
 	}
 	for i := 0; i < mpp.NumField(); i++ {
-		if !reflect.DeepEqual(app.Field(i), mpp.Field(i)) {
-			t.Errorf("mismatch in field %d:\n%#v\n%#v", i, app.Field(i), mpp.Field(i))
+		mf := mpp.Field(i)
```

Comment:
Since the fields are now missing they do not line up by field number.  I am now fetching the fields in the agentpoolprofile from the derived masterpoolprofile.